### ISSUE TITLE
substitutions not consistently set

### ISSUE THREAD

**Issue by terencechow** [sendgrid/sendgrid-nodejs#607]
#### Issue Summary

I am not sure why but my substitutions are not consistently filled out. I'm using node 6.11.4 and `@sendgrid/mail` 6.1.4

In my code I have the following:

emails.js

    import sgMail from '@sendgrid/mail'
    sgMail.setApiKey(config.sendGrid.apiKey)
    sgMail.setSubstitutionWrappers('{{', '}}')

    const defaultOptions = {
        from: 'noreply@mycompany.com',
        bcc: 'dev@mycompany.com',
        mail_settings: { sandbox_mode: { enable: config.sendGrid.sandboxMode } }
    }
    export function email(emailDetails) {
        const messages = emailDetails.map(detail => {
            return { ...defaultOptions, ...detail }
        })
        console.log('messages: ',messages)
        return sgMail.send(messages)
    }

Which is used in my server routes to send emails on different triggers.

When i look at my console logs I see the following for messages:

```
messages: [{ 
   from: 'noreply@mycompany.com',
   bcc: 'dev@mycompany.com',
   mail_settings: { sandbox_mode: [Object] },
   substitutions: { 
      firstName: 'Bob',
      projectName: 'Project 1',
      count: 8,
      deadline: '2018-02-16 05:05:00'
   },
   templateId: '89f98810-REST-OF-ID-REMOVED',
   to:
    { email: 'someuser@mycompany.com',
      name: 'Bob' } 
}]
```

given the above array passed to the sendgrid mail library, why would the substitutions sometimes be filled out and sometimes not be filled out?

**Comment by terencechow** [comment]
It appears that isn't set on the BCC user?

**Comment by terencechow** [comment]
After a long day of hairpulling i think I have figured out the issue.

There is a setting on the sendgrid website that you can enable to 'send a copy to a bcc email'. That was set in my case for `dev@mycompany.com`. When I tried using `sendgrid-nodejs` and included the bcc `dev@mycompany.com`, my best guess is that sendgrid was doing something internally, realizing that the same email was included twice, and removing it. Then, it was sending a copy of the email to the bcc email as per the settings. I think when it sends this copy, it doesnt include any substitutions. 

Sometimes however, my bcc email would get substitutions so I think whether it stripped my bcc email from @sendgrid/mail or whether it stripped it from sending from the settings might be non-deterministic. When it sends with my bcc email it seems to preserve substitutions, but when it sends a copy from the settings, it seems to have no substitutions.

So long story short, I will be setting up a new email tomorrow that is different from the bcc copy of every email required.

If any sendgrid employees could review what I have said and either confirm or tell me i'm crazy that would be great. Theres no other way I can imagine that an email with a bcc sends a different email to the To: email and to the BCC: email. (I.e. the To: user clearly gets an email with substitutions and the BCC: user gets a template.

Long story short, if you come across this problem either disable the setting in your sendgrid ui settings, or if you cant, then bcc a different email. Would love to hear commentary from sendgrid employees if this is the issue

**Comment by thinkingserious** [comment]
Hi @terencechow,

I'm looking internally to find out who has the best knowledge of how the 'send a copy to a bcc email' interacts with the templating function. My guess is that it not designed for that use case.

Please do let us know how your test turns out :)

With Best Regards,

Elmer

**Comment by terencechow** [comment]
I've been told my support that the BCC in the global object does not get any substitutions. 

I've also been told to create a personalizations array of objects with the BCC in each personalization object. That said, the support ticket is still open because now the issue is that a personalization array of objects with say 3 unique emails across 4 people ( To: A, BCC: B and To: C, BCC: A for 2 different emails) results in A not receiving 2 emails but 1 email, which shouldn't be the case... Not sure if it is related to the  'send a copy to a bcc email' but will find out soon hopefully...

**Comment by thinkingserious** [comment]
Thanks for the follow up @terencechow!

Please let us know when you get the final answer.
